### ISSUE TITLE
terraform-provider-mongodbatlas_v1.6.0 plugin crashed

### ISSUE THREAD

**Issue by ppiankov** [mongodb/terraform-provider-mongodbatlas#935]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
❯ terraform version
Terraform v1.3.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.40.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.16.0
+ provider registry.terraform.io/mongodb/mongodbatlas v1.6.0

same was with
Terraform v1.3.3
```

### Terraform Configuration File 

```hcl
terraform {
  required_providers {
    mongodbatlas = {
      source = "mongodb/mongodbatlas"
      version = "1.6.0"
    }
  }
}

resource "mongodbatlas_search_index" "textSearch" {
  name = "textSearch"
  project_id = local.mdb_project_id
  cluster_name = local.mdb_cluster_main

  analyzer = "lucene.standard"
  collection_name = "offers"
  database = "offers-test"
  mappings_dynamic = false
  mappings_fields = <<-EOF
{
  "brand": {
    "fields": {
      "name": {
        "analyzer": "lucene.standard",
        "searchAnalyzer": "lucene.standard",
        "type": "string",
        "norms": "omit",
      }
    },
    "type": "document"
  },
  "settings": {
    "fields": {
      "description": {
        "analyzer": "lucene.standard",
        "searchAnalyzer": "lucene.standard",
        "type": "string",
        "norms": "omit"
      },
      "title": {
        "analyzer": "lucene.standard",
        "searchAnalyzer": "lucene.standard",
        "type": "string",
        "norms": "omit"
      }
    },
    "type": "document"
  }
}
EOF

  search_analyzer = "lucene.standard"
}
```

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Expected Behavior
the existing index should get updated

### Actual Behavior
crash dump

### Crash Output
**https://gist.github.com/ppiankov/2ed7249c41f942bdf9f179fd2e8bc3fa**

**Comment by evertsd** [comment]
@ppiankov Based on this, it looks like an issue with the provider not being able to handle going from a non-nil value of `search_analyzer` to value `nil`.

A potential workaround *may* be adding `search_analyzer = ""` to your `mongodbatlas_search_index.textSearch` resource

**Comment by ppiankov** [comment]
thanks search_analyzer = "" helped.

**Comment by ppiankov** [comment]
but after this fix the index is not working at all

**Comment by evertsd** [comment]
Did you still need the following?
```
search_analyzer = "lucene.standard"
```

Or does including the above cause the crash output in the gist?

**Comment by ppiankov** [comment]
yes if we include 

```
search_analyzer = "lucene.standard"
```

it will crash.

and this is how a healthy working index created manually looks like
```
{
  "analyzer": "lucene.standard",
  "mappings": {
    "dynamic": false,
    "fields": {
      "brand": {
        "fields": {
          "name": {
            "analyzer": "lucene.standard",
            "norms": "omit",
            "searchAnalyzer": "lucene.standard",
            "type": "string"
          }
        },
        "type": "document"
      },
      "settings": {
        "fields": {
          "description": {
            "analyzer": "lucene.standard",
            "norms": "omit",
            "searchAnalyzer": "lucene.standard",
            "type": "string"
          },
          "title": {
            "analyzer": "lucene.standard",
            "norms": "omit",
            "searchAnalyzer": "lucene.standard",
            "type": "string"
          }
        },
        "type": "document"
      }
    }
```

**Comment by ppiankov** [comment]
it does not crash if no index pre-exist and it creates a new one. it crashes only when there is an existing index. 
Also even when it creates the index from scratch the index does not work as intended.

**Comment by evertsd** [comment]
Thanks for reporting @ppiankov, the issue has been identified and we're planning to fix in the next release. [INTMDB-341](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/964) internal engineering ticket.

**Comment by Zuhairahmed** [comment]
@ppiankov brief update, we have allocated this fix to our next v1.8.0 release which should be out next week. will keep you posted when it has been released.

**Comment by Zuhairahmed** [comment]
@ppiankov we just released v1.8.0, feel free to give it try! issue should have been resolved.

**Comment by ppiankov** [comment]
works. thanks.
